core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +539,48 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating the images in the exploration
+    from the GAE to GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = (
+            file_snapshot_content_model.get_unversioned_instance_id())
+        filetype = instance_id[instance_id.rfind('.') + 1:]
+        # to filter all the image entries.
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            pattern = re.compile(
+                r"^/([^/]+)/assets/([^/]+)\.(png|jpg|gif|jpeg)$")
+            parts_of_id = pattern.match(instance_id)
+            if not parts_of_id:
+                yield('Error: The instance_id is not correct', instance_id)
+            else:
+                filename = parts_of_id.group(2) + '.' + parts_of_id.group(3)
+                filepath = 'assets/' + filename
+                exploration_id = parts_of_id.group(1)
+                file_model = file_models.FileModel.get_model(
+                    exploration_id, filepath, False)
+                if file_model:
+                    if file_model.deleted:
+                        yield('Error: found deleted file', file_model.id)
+                    else:
+                        # content = file_model.content
```

Comment:
This PR cannot be submitted (or, in fact, run) until there is no commented-out code.

Reply:
Adding to the GCS part is commented out. Because I can't test it locally and still figuring out a way to do it.

Reply:
Hi @ishucr7  Can you describe the issue you are facing while testing with a few screenshots if possible, may be I can help you then, Thanks!

Reply:
Yeah sure. I am able to get the server up, but on running the job on the admin page gives the following error 
![job_error](https://user-images.githubusercontent.com/23234392/42190263-b6866cce-7e79-11e8-887d-e626f085e0a2.png)

The error on the console is 
![er1](https://user-images.githubusercontent.com/23234392/42190348-354de672-7e7a-11e8-9534-7771c5d442b4.png)
![Uploading er2.png…]()
![er3](https://user-images.githubusercontent.com/23234392/42190346-34c469b0-7e7a-11e8-90fa-38c550324a34.png)